test/__tests__/__integration_util_test__/flattenJSONUtil.test.js

```diff
@@ -0,0 +1,553 @@
+  "products[2].url": "https://www.example.com/product/offer-t-shirt",
+  "products[2].value": 12.99,
+  "products[2].variant": "Black",
+  revenue: 31.98,
+  shipping: 4,
+  value: 31.98,
+  "originalArray[0].nested_field": "nested value",
+  "originalArray[0].tags[0]": "tag_1",
+  "originalArray[0].tags[1]": "tag_2",
+  "originalArray[0].tags[2]": "tag_3",
+  "originalArray[1].nested_field": "nested value",
+  "originalArray[1].tags[0]": "tag_1",
+  "originalArray[2].nested_field": "nested value",
+  category: "some category"
+};
+
+const PayloadInput4 = {
+  timezone: "Asia/Kolkata",
+  app: {
+    build: "1.0.0",
+    name: "RudderLabs JavaScript SDK",
+    namespace: "com.rudderlabs.javascript",
+    version: "1.1.6"
+  },
+  library: {
+    name: "RudderLabs JavaScript SDK",
+    version: "1.1.6"
+  },
+  locale: "en-GB",
+  os: {
+    name: "",
+    version: ""
+  },
+  page: {
+    path: "/testing/script-test.html",
+    referrer: "",
+    search: "",
+    title: "",
+    url: "http://localhost:3243/testing/script-test.html"
+  },
+  screen: {
+    density: 2
+  },
+  traits: {
+    company: {
+      id: "abc123"
+    },
+    createdAt: "Thu Mar 24 2016 17:46:45 GMT+0000 (UTC)",
+    email: "ruchira@gmail.com",
+    name: "Ruchira Moitra",
+    plan: "Enterprise"
+  },
+  userAgent:
+    "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.80 Safari/537.36"
+};
+
+const PayloadOutput4 = {
+  "app.build": "1.0.0",
+  "app.name": "RudderLabs JavaScript SDK",
+  "app.namespace": "com.rudderlabs.javascript",
+  "app.version": "1.1.6",
+  "library.name": "RudderLabs JavaScript SDK",
+  "library.version": "1.1.6",
+  locale: "en-GB",
+  "os.name": "",
+  "os.version": "",
+  "page.path": "/testing/script-test.html",
+  "page.referrer": "",
+  "page.search": "",
+  "page.title": "",
+  "page.url": "http://localhost:3243/testing/script-test.html",
+  "screen.density": 2,
+  timezone: "Asia/Kolkata",
+  "traits.company.id": "abc123",
+  "traits.createdAt": "Thu Mar 24 2016 17:46:45 GMT+0000 (UTC)",
+  "traits.email": "ruchira@gmail.com",
+  "traits.name": "Ruchira Moitra",
+  "traits.plan": "Enterprise",
+  userAgent:
+    "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.80 Safari/537.36"
+};
+
+describe("flattenJSon util test", () => {
+  it("Test Case 1", () => {
+    const outputArray = utilObj[fnName](PayloadInput1);
+    expect(outputArray).toEqual(PayloadOutput1);
+  });
+
+  it("Test Case 2", () => {
+    const outputArray = utilObj[fnName](PayloadInput2);
+    expect(outputArray).toEqual(PayloadOutput2);
+  });
+
+  it("Test Case 3", () => {
+    const outputArray = utilObj[fnName](PayloadInput3);
+    expect(outputArray).toEqual(PayloadOutput3);
+  });
+
+  it("Test Case 4", () => {
+    const outputArray = utilObj[fnName](PayloadInput4);
+    expect(outputArray).toEqual(PayloadOutput4);
+  });
+});
```

Comment:
We can format this file in a different way. One folder should contain two files, one is this test file and another is the input-output file. And in this test file, we should introduce an iterator apart from statically writing each case. In this way, it will be easier to add new test cases if required.